run_test.sh

```diff
@@ -244,12 +257,15 @@ case "${os_id}" in
     sles | suse | opensuse)
         expected_dest_loc=updates
         ;;
-    arch | debian | ubuntu | linuxmint)
+    arch)
+        expected_dest_loc=updates/dkms
+        mod_compression_ext=
```

Comment:
Arch uses zstd as the default compression for some time now: https://archlinux.org/news/moving-to-zstandard-images-by-default-on-mkinitcpio/

So I guess we shouldn't reset it here?

Reply:
Just tested it on my Arch system:
```
$ grep "CONFIG_MODULE_COMPRESS_" /lib/modules/6.6.4-arch1-1/build/.config
# CONFIG_MODULE_COMPRESS_NONE is not set
# CONFIG_MODULE_COMPRESS_GZIP is not set
# CONFIG_MODULE_COMPRESS_XZ is not set
CONFIG_MODULE_COMPRESS_ZSTD=y
```

Reply:
But apparently they don't use the the .ko.zst extension otherwise the testsuite wouldn't have failed after finding CONFIG_MODULE_COMPRESS_ZSTD=y (https://github.com/dell/dkms/actions/runs/7343241921/job/19993570808). (And Centos and friends use .ko.gz without setting CONFIG_MODULE_COMPRESS_GZIP=y.)

Reply:
They do on my (pretty much default) system:
```
$ find /usr/lib/modules/6.6.4-arch1-1/ -name "*.zst"
/usr/lib/modules/6.6.4-arch1-1/kernel/virt/lib/irqbypass.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/mm/hwpoison-inject.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/kernel/kheaders.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/objagg.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/libcrc32c.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/math/cordic.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/parman.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/polynomial.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crc7.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crypto/libarc4.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crypto/libchacha.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crypto/gf128mul.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crypto/libchacha20poly1305.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crypto/libpoly1305.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crypto/libcurve25519-generic.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crypto/libdes.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crypto/libcurve25519.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/bch.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/crc4.ko.zst
/usr/lib/modules/6.6.4-arch1-1/kernel/lib/lru_cache.ko.zst
[...]
```

Reply:
Having the kernel image installed sounds like a more realistic test environment. This is what normal users will have.

Parsing modules.dep to detect the compression is ... suboptimal. But there is already #319